Fixes #3443

bikallem opened the pull request:
Fixes #3443

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
strict_package_deps doesn't understand transitive deps

As found out by @talex5

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
strict_package_deps doesn't understand transitive deps

As found out by @talex5

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
strict_package_deps doesn't understand transitive deps

As found out by @talex5

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Reproduce include_subdirs + cinaps bug

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

thizanne committed:
Fix small typo in doc

Signed-off-by: Thibault Suzanne <thi.suzanne@gmail.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

thizanne committed:
Fix small typo in doc

Signed-off-by: Thibault Suzanne <thi.suzanne@gmail.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

thizanne committed:
Fix small typo in doc

Signed-off-by: Thibault Suzanne <thi.suzanne@gmail.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Remove stdune from configurator

Necessary functions are moved to Import

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Remove stdune from configurator

Necessary functions are moved to Import

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Remove stdune from configurator

Necessary functions are moved to Import

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Add signature to Action_unexapnded.Infer

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Add applicative interface to build

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Simplify flag handling in preprocessing

There was some common code between linting and normal preprocessing that
can be factored out

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Fix #3399

There's no need to pass `-ocamlc` manually

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Write mli for Action_ast

This module was missing an mli for quite a while. The omission must have
been accidental.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Stop passing external lib deps mode

It's not set per context (or super context), but for an entire
subcommand.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Use csexp to communicate with configurator

This allows us to drop the dune_lang dependency

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Use csexp to communicate with configurator

This allows us to drop the dune_lang dependency

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Use csexp to communicate with configurator

This allows us to drop the dune_lang dependency

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Drop stdune dependency in configurator

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Drop stdune dependency in configurator

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Drop stdune dependency in configurator

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

snowleopard committed:
Drop partial support for hash collisions in Dune cache

See #3357.

Signed-off-by: Andrey Mokhov <amokhov@janestreet.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

snowleopard committed:
Refactor [FirstTwoCharsSubdir.list]

Use [Path.readdir_unsorted] which is more type-safe and is
explicit about errors. Also, simplify [is_hex_char].

Signed-off-by: Andrey Mokhov <amokhov@janestreet.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

voodoos committed:
Add a test illustrating the issue.

Signed-off-by: Ulysse Gérard <thevoodoos@gmail.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

snowleopard committed:
Remove full paths to file storage from cache metadata

See #3357.

Signed-off-by: Andrey Mokhov <amokhov@janestreet.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

snowleopard committed:
Simplify the test on unexpected directories in cache root

Signed-off-by: Andrey Mokhov <amokhov@janestreet.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

voodoos committed:
Enhance "since" error message

Signed-off-by: Ulysse Gérard <thevoodoos@gmail.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

voodoos committed:
Update tests with new error messages

Signed-off-by: Ulysse Gérard <thevoodoos@gmail.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Document Compilation_context.opaque

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3400 from rgrinberg/fix-quick-start

Fix #3399
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3397 from rgrinberg/applicative-build

Add applicative interface to build
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3396 from rgrinberg/pp-lint-refactor

Simplify flag handling in preprocessing
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3401 from rgrinberg/action-mli

Write mli for Action_ast
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3404 from rgrinberg/external-lib-deps-mode

Stop passing external lib deps mode
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3414 from voodoos/issue-3412

Better "since" error messages
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

snowleopard committed:
Add support for earlier message protocol versions

Signed-off-by: Andrey Mokhov <amokhov@janestreet.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Organize install_rules a little better

* Move definitions close to their usage
* Do not leak unnecessary names
* Add some comments

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3422 from rgrinberg/reorg-install-rules

Organize install_rules a little better
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Package dependency handling

* Properly error on missing packages in (package ..) definitions
* Move <pkg>-files aliases to the package dir. This makes rule loading
lazier.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Package dependency handling

* Properly error on missing packages in (package ..) definitions
* Move <pkg>-files aliases to the package dir. This makes rule loading
lazier.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Package dependency handling

* Properly error on missing packages in (package ..) definitions
* Move <pkg>-files aliases to the package dir. This makes rule loading
lazier.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Add Fiber.Result module

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3430 from rgrinberg/add-fiber-result

Add Fiber.Result module
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3241 from rgrinberg/include-subdirs-cinaps

Reproduce include_subdirs + cinaps bug
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3363 from thizanne/doc_typo

Fix small typo in doc
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
promote tests

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
promote tests

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
promote tests

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3165 from rgrinberg/strict-package-deps-transitive

strict_package_deps doesn't understand transitive deps
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3424 from rgrinberg/package-dep-handling

Package dependency handling
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Revert "Add Fiber.Result module"

This reverts commit e61f1f4cbc12fe6081c07c4a353cca61c3c030dc.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Revert "Add Fiber.Result module"

This reverts commit e61f1f4cbc12fe6081c07c4a353cca61c3c030dc.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Revert "Add Fiber.Result module"

This reverts commit e61f1f4cbc12fe6081c07c4a353cca61c3c030dc.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Drop ocaml_config dependency from Stdune

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Drop ocaml_config dependency from Stdune

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Drop ocaml_config dependency from Stdune

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Remove dune_lang as a dependency

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Remove dune_lang as a dependency

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Remove dune_lang as a dependency

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Add unix dependency to configurator

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Add unix dependency to configurator

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Add unix dependency to configurator

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Fixes ##3443

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Fixes ##3443

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Fixes ##3443

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Fixes ##3443

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Fixes ##3443

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Fixes ##3443

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

hhugo committed:
Fix META generation wrt javascript deps

Signed-off-by: Hugo Heuzard <hugo.heuzard@gmail.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

hhugo committed:
Fix META generation wrt javascript deps

Signed-off-by: Hugo Heuzard <hugo.heuzard@gmail.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

hhugo committed:
Fix META generation wrt javascript deps

Signed-off-by: Hugo Heuzard <hugo.heuzard@gmail.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

hhugo committed:
Fix META generation wrt javascript deps

Signed-off-by: Hugo Heuzard <hugo.heuzard@gmail.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

hhugo committed:
Fix META generation wrt javascript deps

Signed-off-by: Hugo Heuzard <hugo.heuzard@gmail.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

toots committed:
Update fcntl_stubs.c

Signed-off-by: Romain Beauxis <toots@rastageeks.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

toots committed:
Update fcntl_stubs.c

Signed-off-by: Romain Beauxis <toots@rastageeks.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

toots committed:
Update fcntl_stubs.c

Signed-off-by: Romain Beauxis <toots@rastageeks.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

toots committed:
Update fcntl_stubs.c

Signed-off-by: Romain Beauxis <toots@rastageeks.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

toots committed:
Update fcntl_stubs.c

Signed-off-by: Romain Beauxis <toots@rastageeks.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Add include_dirs field to dune describe

This field contains the list of directories containing all the cmi's of
a library.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Add include_dirs field to dune describe

This field contains the list of directories containing all the cmi's of
a library.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Add include_dirs field to dune describe

This field contains the list of directories containing all the cmi's of
a library.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3445 from hhugo/fix-meta-gen

Fix META generation wrt javascript deps
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

jeremiedimino committed:
Insert a constraint on the version of dune (#3434)

Even if the dependency is explicitly listed in the dune-project file

Fixes #3427

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

jeremiedimino committed:
Insert a constraint on the version of dune (#3434)

Even if the dependency is explicitly listed in the dune-project file

Fixes #3427

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

jeremiedimino committed:
Insert a constraint on the version of dune (#3434)

Even if the dependency is explicitly listed in the dune-project file

Fixes #3427

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

jeremiedimino committed:
Insert a constraint on the version of dune (#3434)

Even if the dependency is explicitly listed in the dune-project file

Fixes #3427

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

jeremiedimino committed:
Insert a constraint on the version of dune (#3434)

Even if the dependency is explicitly listed in the dune-project file

Fixes #3427

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

jeremiedimino commented:
Thanks. Could you add a changelog entry?

jeremiedimino commented on bin/init.ml:
```diff
@@ -30,17 +30,14 @@ let atom_printer ppf a = Format.pp_print_string ppf (Dune_lang.Atom.to_string a)
 let component_name_parser s =
```
@shonfeder why is this called "component" rather than "library"?

shonfeder replied:
iirc, this is because it should parse names for any "component", including executables, projects, and tests. The idea being that dune build many kinds of components -- not just libraries -- and the `init` should (in theory) be extensible to initialize any of them.

jeremiedimino replied:
Ah I see, that makes sense. Thanks! This probably deserve a comment, otherwise the usage of the `Lib_name....` tends to indicate that this is for parsing library names.

jeremiedimino committed:
Remove Fiber.Once (#3448)

And re-implement its only use case using simpler concepts.

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

jeremiedimino committed:
Remove Fiber.Once (#3448)

And re-implement its only use case using simpler concepts.

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

jeremiedimino committed:
Remove Fiber.Once (#3448)

And re-implement its only use case using simpler concepts.

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

jeremiedimino committed:
Remove Fiber.Once (#3448)

And re-implement its only use case using simpler concepts.

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

jeremiedimino committed:
Remove Fiber.Once (#3448)

And re-implement its only use case using simpler concepts.

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

shonfeder commented on bin/init.ml:
```diff
@@ -30,17 +30,14 @@ let atom_printer ppf a = Format.pp_print_string ppf (Dune_lang.Atom.to_string a)
-      [ Pp.textf "invalid component name `%s'" s
-      ; Lib_name.Local.valid_format_doc
-      ]
+    User_error.make [ Pp.textf "invalid component name `%s'" s ]
```
To be user friendly, shouldn't we still give some feedback here about what counts as a valid component name?

bikallem replied:
FWIU, There is no validation logic for ```Lib_name.of_string_opt```, i.e. it just creates ```Some s``` always, so there is no error message to display. Perhaps once this is decided - i.e. what constitutes as a valid lib name - we can revisit this again. 

As a matter of fact since we are never going to get any error here, we can remove the rest of the ```libs``` error handling code. But I want the PR to focus on solving a specific bug. The refactoring can come in another PR.

shonfeder replied:
Oh.. well, this verification was added to address #3046 I haven't had time to check this change (or to properly review it) to see the behavior, but I'm wondering how it avoids introducing a regression here...

shonfeder replied:
So, as I feared, this does introduce a regression. If I run

```sh
dune build @dune-init
```

we can see that invalid module names are no longer being caught: 

```
Done: 3926/3928 (jobs: 1)File "test/blackbox-tests/test-cases/dune-init/run.t", line 1, characters 0-0:
         git (internal) (exit 1)
(cd _build/default && /usr/bin/git diff --no-index --color=always -u test/blackbox-tests/test-cases/dune-init/run.t test/blackbox-tests/test-cases/dune-init/run.t.corrected)
diff --git a/test/blackbox-tests/test-cases/dune-init/run.t b/test/blackbox-tests/test-cases/dune-init/run.t.corrected
index 1f92984b..c3271187 100644
--- a/test/blackbox-tests/test-cases/dune-init/run.t
+++ b/test/blackbox-tests/test-cases/dune-init/run.t.corrected
@@ -239,13 +239,7 @@ Comments in dune files are preserved
 Will not create components with invalid names
 
   $ dune init lib invalid-component-name ./_test_lib
-  dune: NAME argument: invalid component name `invalid-component-name'
-        Library names must be non-empty and composed only of the
-        following
-        characters: 'A'..'Z', 'a'..'z', '_' or '0'..'9'.
-  Usage: dune init [OPTION]... INIT_KIND NAME [PATH]
-  Try `dune init --help' or `dune --help' for more information.
-  [1]
+  Success: initialized library component named invalid-component-name
   $ test -f ./_test_lib
   [1]
```

I think a proper fix for this will need to a bit more subtle. I should have time to take a look tomorrow evening, and can provide some pointers on what I think might work, if that would be helpful!

bikallem replied:
Commit https://github.com/ocaml/dune/pull/3444/commits/831900b9f1c34d2dd5aea1e4616233c67bf4b8a8 addresses test failure now. 

I have also made changes to the error mesage to be more specific now.

bikallem committed:
Merge branch 'master' into master
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Merge branch 'master' into master

bikallem committed:
add to CHANGES.md

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
add to CHANGES.md

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
add to CHANGES.md

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
add to CHANGES.md

bikallem committed:
add to CHANGES.md

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
add to CHANGES.md

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
add to CHANGES.md

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
add to CHANGES.md

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem commented:
Added CHANGES.md entry as requested.

jeremiedimino commented on CHANGES.md:
```diff
@@ -1,5 +1,7 @@
 next
 ----
+- Fix a bug where valid lib names in `dune init exec --libs=lib1,lib2` 
+  results in error. (#3444, fix #3443, @bikallem)
```
```suggestion
  result in error. (#3444, fix #3443, @bikallem)
```

jeremiedimino commented:
@shonfeder do you mind taking over the review and merging of this PR? I'm not super familiar with this part of the codebase.

bikallem committed:
Update CHANGES.md

Co-authored-by: Jérémie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Update CHANGES.md

Co-authored-by: Jérémie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Update CHANGES.md

Co-authored-by: Jérémie Dimino <jeremie@dimino.org>

bikallem committed:
Update CHANGES.md

Co-authored-by: Jérémie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Update CHANGES.md

Co-authored-by: Jérémie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Update CHANGES.md

Co-authored-by: Jérémie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Update CHANGES.md

Co-authored-by: Jérémie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Update CHANGES.md

Co-authored-by: Jérémie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3450 from rgrinberg/dune-describe-include-dirs

Add include_dirs field to dune describe
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3410 from rgrinberg/configurator-remove-stdune

Remove stdune from configurator and switch to csexp
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

greedy committed:
Fix for #2990

The second argument of a 'diff?' action is removed from the set of
targets as the action consumes (i.e. deletes) the file.

Signed-off-by: Geoffrey Reedy <geoff@programmer-monk.net>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

greedy committed:
Fix for #2990

The second argument of a 'diff?' action is removed from the set of
targets as the action consumes (i.e. deletes) the file.

Signed-off-by: Geoffrey Reedy <geoff@programmer-monk.net>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

greedy committed:
Fix for #2990

The second argument of a 'diff?' action is removed from the set of
targets as the action consumes (i.e. deletes) the file.

Signed-off-by: Geoffrey Reedy <geoff@programmer-monk.net>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Refine the type of the diff action

The second argument must be in the build dir, as we cannot produce
targets outside the build dir.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Refine the type of the diff action

The second argument must be in the build dir, as we cannot produce
targets outside the build dir.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Refine the type of the diff action

The second argument must be in the build dir, as we cannot produce
targets outside the build dir.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

greedy committed:
reproduce 2990

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

greedy committed:
reproduce 2990

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

greedy committed:
reproduce 2990

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

shonfeder commented:
> @shonfeder do you mind taking over the review and merging of this PR? I'm not super familiar with this part of the codebase.

Not at all -- Happy to do so. Sorry I didn't notice when the issue was filed :)

rgrinberg committed:
Merge pull request #3457 from greedy/fix-2990

Fix for #2990
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

jeremiedimino committed:
Extend the test case for #2990

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

jeremiedimino committed:
Extend the test case for #2990

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

jeremiedimino committed:
Extend the test case for #2990

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
re-add original component-name validation functionality. Parse --libs as we parse --ppx

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
re-add original component-name validation functionality. Parse --libs as we parse --ppx

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
re-add original component-name validation functionality. Parse --libs as we parse --ppx

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
re-add original component-name validation functionality. Parse --libs as we parse --ppx

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Merge branch 'master' into master

bikallem committed:
Merge branch 'master' into master
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

stephanieyou committed:
Add support for bisect_ppx (#3403)

Signed-off-by: Stephanie You <youstephanie98@gmail.com>
Co-authored-by: Jérémie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

stephanieyou committed:
Add support for bisect_ppx (#3403)

Signed-off-by: Stephanie You <youstephanie98@gmail.com>
Co-authored-by: Jérémie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

stephanieyou committed:
Add support for bisect_ppx (#3403)

Signed-off-by: Stephanie You <youstephanie98@gmail.com>
Co-authored-by: Jérémie Dimino <jeremie@dimino.org>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
merge master

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
merge master

jeremiedimino commented:
> Not at all -- Happy to do so. Sorry I didn't notice when the issue was filed :)

Thanks and no problem! Moving forward, I'll assign to you the issues related to `dune init` when I see them. The rate of new issues/PRs is now high enough and I expect that not everyone looks at every single new issue or PR.

bikallem committed:
Update CHANGES.md

bikallem committed:
Update CHANGES.md

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Update CHANGES.md
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Update CHANGES.md

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Merge branch 'master' into master

bikallem committed:
Merge branch 'master' into master
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
sync CHANGES.md

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
sync CHANGES.md

Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
reproduce #3440

$ `dune rules` stopped working. This commit just adds a test witnessing
this bug.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Fix #3440

Path.of_string should be called after the build directory was set.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

rgrinberg committed:
Merge pull request #3459 from rgrinberg/3340

Fix #3440
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Merge branch 'master' into master

bikallem committed:
Merge branch 'master' into master
Signed-off-by: Bikal Gurung <gbikal@gmail.com>

bikallem committed:
Merge branch 'master' of https://github.com/bikallem/dune

bikallem commented:
@shonfeder I will create a new PR with the required changes since the commit history here has message up due to some git rebase mistakes owing to DCO failures.
